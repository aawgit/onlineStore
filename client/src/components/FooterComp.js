/* eslint-disable jsx-a11y/anchor-is-valid */
import React from 'react';

const FooterComp = () => {
	const top = () => {
		document.body.scrollTop = 0; // For Safari
		document.documentElement.scrollTop = 0; // For Chrome, Firefox, IE and Opera
	};
	return (
		<div>
			<footer className='text-muted'>
				<div className='container'>
					<p className='float-right'>
						<a href='#' onClick={top}>
							Back to top
						</a>
					</p>
					<p>Develped by &copy; AW</p>
				</div>
			</footer>
		</div>
	);
};
export default FooterComp;
