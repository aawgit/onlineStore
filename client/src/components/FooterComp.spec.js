import React from 'react';
import { mount } from 'enzyme';
import FooterComp from './FooterComp';

describe('<FooterComp />', () => {
	let wrapper;

	beforeEach(() => {
		wrapper = mount(<FooterComp />);
	});

	it('should render', () => {
		expect(wrapper).toMatchSnapshot();
	});

	describe('Back to Top button', () => {
		it('should scroll top', () => {
			expect(document.body.scrollTop).toEqual(0);
			document.body.scrollTop = 100;
			wrapper.find('#scrollToTop').simulate('click');
			expect(document.body.scrollTop).toEqual(0);
		});
	});
});
