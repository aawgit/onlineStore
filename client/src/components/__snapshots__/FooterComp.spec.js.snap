// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FooterComp /> should render 1`] = `
ReactWrapper {
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "getWrappingComponentRenderer": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
    "wrapInvoke": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {},
    "ref": null,
    "rendered": Object {
      "instance": <footer
        class="text-muted fixed-bottom"
      >
        <div
          class="container"
        >
          <p
            class="float-right"
          >
            <a
              href="#"
              id="scrollToTop"
            >
              Back to top
            </a>
          </p>
          <p>
            Develped by © AW
          </p>
        </div>
      </footer>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        >
          <p
            className="float-right"
          >
            <a
              href="#"
              id="scrollToTop"
              onClick={[Function]}
            >
              Back to top
            </a>
          </p>
          <p>
            Develped by © AW
          </p>
        </div>,
        "className": "text-muted fixed-bottom",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <div
            class="container"
          >
            <p
              class="float-right"
            >
              <a
                href="#"
                id="scrollToTop"
              >
                Back to top
              </a>
            </p>
            <p>
              Develped by © AW
            </p>
          </div>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="float-right"
              >
                <a
                  href="#"
                  id="scrollToTop"
                  onClick={[Function]}
                >
                  Back to top
                </a>
              </p>,
              <p>
                Develped by © AW
              </p>,
            ],
            "className": "container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <p
                class="float-right"
              >
                <a
                  href="#"
                  id="scrollToTop"
                >
                  Back to top
                </a>
              </p>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  href="#"
                  id="scrollToTop"
                  onClick={[Function]}
                >
                  Back to top
                </a>,
                "className": "float-right",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": <a
                    href="#"
                    id="scrollToTop"
                  >
                    Back to top
                  </a>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Back to top",
                    "href": "#",
                    "id": "scrollToTop",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Back to top",
                  ],
                  "type": "a",
                },
              ],
              "type": "p",
            },
            Object {
              "instance": <p>
                Develped by © AW
              </p>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Develped by © AW",
              },
              "ref": null,
              "rendered": Array [
                "Develped by © AW",
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "footer",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": Object {
        "instance": <footer
          class="text-muted fixed-bottom"
        >
          <div
            class="container"
          >
            <p
              class="float-right"
            >
              <a
                href="#"
                id="scrollToTop"
              >
                Back to top
              </a>
            </p>
            <p>
              Develped by © AW
            </p>
          </div>
        </footer>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="container"
          >
            <p
              className="float-right"
            >
              <a
                href="#"
                id="scrollToTop"
                onClick={[Function]}
              >
                Back to top
              </a>
            </p>
            <p>
              Develped by © AW
            </p>
          </div>,
          "className": "text-muted fixed-bottom",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <div
              class="container"
            >
              <p
                class="float-right"
              >
                <a
                  href="#"
                  id="scrollToTop"
                >
                  Back to top
                </a>
              </p>
              <p>
                Develped by © AW
              </p>
            </div>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="float-right"
                >
                  <a
                    href="#"
                    id="scrollToTop"
                    onClick={[Function]}
                  >
                    Back to top
                  </a>
                </p>,
                <p>
                  Develped by © AW
                </p>,
              ],
              "className": "container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <p
                  class="float-right"
                >
                  <a
                    href="#"
                    id="scrollToTop"
                  >
                    Back to top
                  </a>
                </p>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    href="#"
                    id="scrollToTop"
                    onClick={[Function]}
                  >
                    Back to top
                  </a>,
                  "className": "float-right",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": <a
                      href="#"
                      id="scrollToTop"
                    >
                      Back to top
                    </a>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Back to top",
                      "href": "#",
                      "id": "scrollToTop",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": Array [
                      "Back to top",
                    ],
                    "type": "a",
                  },
                ],
                "type": "p",
              },
              Object {
                "instance": <p>
                  Develped by © AW
                </p>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Develped by © AW",
                },
                "ref": null,
                "rendered": Array [
                  "Develped by © AW",
                ],
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "footer",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__linkedRoots__): Array [],
  Symbol(enzyme.__unrendered__): <FooterComp />,
  Symbol(enzyme.__updatedBy__): null,
}
`;
